build/Linux+BSD/portable/x86_64/Recipe

```diff
@@ -1,92 +1,100 @@
 #!/bin/bash
 
-# This script should be run in an Ubuntu 12.04 "Precise Pangolin" Docker image
+# For maximum AppImage compatibility, run this script on the oldest Ubuntu
+# that is still supported by Canonical (ideally the one used on Travis CI).
 
 set -e # Halt on errors
 set -x # Be verbose
 
+# go one-up from MuseScore root dir regardless of where script was run from:
+cd "$(dirname "$(readlink -f "${0}")")/../../../../.."
+
 ##########################################################################
 # GET DEPENDENCIES
 ##########################################################################
 
-# go one-up from MuseScore root dir regardless of where script was run from:
-cd "$(dirname "$(readlink -f "${0}")")/../../../../.."
+# DISTRIBUTION PACKAGES
+
+# These are installed by default on Travis CI, but not on Docker
+apt_packages_basic=(
+  # alphabetical order please
+  git
+  pkg-config
+  software-properties-common # installs `add-apt-repository`
+  unzip
+  )
+
+# These are the same as on Travis CI
+apt_packages_standard=(
+  # alphabetical order please
+  # cmake # using more recent upstream version from cmake.org
+  curl
+  libasound2-dev # ALSA
+  libfontconfig1-dev
+  libfreetype6-dev
+  libgl1-mesa-dev
+  libjack-dev
+  libmp3lame-dev
+  libnss3-dev
+  libportmidi-dev
+  libpulse-dev
+  libsndfile1-dev
+  make
+  portaudio19-dev
+  wget
+  )
+
+# MuseScore compiles without these but won't run without them
+apt_packages_runtime=(
+  # alphabetical order please
+  libegl1-mesa-dev
+  libodbc1
+  libpq-dev
+  libxcomposite-dev
+  libxcursor-dev
+  libxi-dev
+  libxkbcommon-x11-0
+  libxtst-dev
+  )
 
 apt-get update # no package lists in Docker image
+apt-get install -y "${apt_packages_basic[@]}" "${apt_packages_standard[@]}" "${apt_packages_runtime[@]}"
 
-# basic dependencies (needed by Docker but provided on Travis)
-apt-get -y install \
-  git \
-  unzip \
-  libxslt1-dev \
-  libxrender-dev \
-  libxcomposite-dev \
-  libdrm-dev \
-  libgl1-mesa-dev \
-  libopenal1 \
-  gstreamer0.10-plugins-base \
-  libmirclient9 \
-  software-properties-common # installs `add-apt-repository`
+# NON-DISTRIBUTION PACKAGES
+# We use these when the default version provided by the distribution is
+# too old or has unwelcome modifications compared to the upstream version.
 
 # get newer compiler
-if [ ! -f /etc/apt/sources.list.d/ubuntu-toolchain-r-test-precise.list ]; then
+if [[ ! -f /etc/apt/sources.list.d/ubuntu-toolchain-r-test-*.list ]]; then
   add-apt-repository -y ppa:ubuntu-toolchain-r/test
   apt-get update
 fi
+apt-get install -y g++-7
```

Comment:
@anatoly-os, I was working on this PR at the same time as your changes in 2ebb47ff4c. Let me know which compiler you want to use. Either seems to work. though it makes sense to use the same as on Travis I guess.